File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> dict;
        dict[0]=1;
        int prefix_sum=0,n=nums.size();
        int ans=0;
        for(int i=0;i<n;i++){
            prefix_sum+=nums[i];
            if(dict.count(prefix_sum-k)){
                ans+=dict[prefix_sum-k];
            }
            dict[prefix_sum]++;
        }
        return ans;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        key_cnt = {} 
        curr_sum = 0 
        key_cnt[0]=1
        total = 0
        for val in nums: 
            curr_sum+=val 
            if curr_sum-k in key_cnt: 
                total+=key_cnt[curr_sum-k]
            key_cnt[curr_sum]=key_cnt.get(curr_sum,0)+1

        return total 